### PR TITLE
[CARBONDATA-2552]Fix Data Mismatch for Complex Data type Array of Timestamp with Dictionary Include

### PR DESCRIPTION
**Pull request by Indhumathi27**
Fix Data Mismatch for Complex Data type Array and Struct of Timestamp with Dictionary Include

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
         NA
 - [x] Any backward compatibility impacted?
         NA
 - [x] Document update required?
          NA
 - [x] Testing done
          NA
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
           NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6143/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4980/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5128/

File: core/src/main/java/org/apache/carbondata/core/scan/complextypes/PrimitiveQueryType.java [hunk 1]
```diff
@@ -95,7 +95,7 @@ public PrimitiveQueryType(String name, String parentname, int blockIndex,
       DimensionRawColumnChunk[] rawColumnChunks, int rowNumber,
       int pageNumber, DataOutputStream dataOutputStream) throws IOException {
     byte[] currentVal = copyBlockDataChunk(rawColumnChunks, rowNumber, pageNumber);
-    if (!this.isDictionary) {
+    if (!this.isDictionary && !this.isDirectDictionary) {
```
Comment (sounakr) [inline core/src/main/java/org/apache/carbondata/core/scan/complextypes/PrimitiveQueryType.java]: Please add test cases corresponding to the changes.

Comment (Indhumathi27) [reply core/src/main/java/org/apache/carbondata/core/scan/complextypes/PrimitiveQueryType.java]: okay

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4993/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5136/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6156/

**Comment by kumarvishal09** [comment]
LGTM

**Pull request closed without merging**